vividus-plugin-azure-resource-manager/src/main/java/org/vividus/azure/configuration/AzureKeyVaultPropertiesProcessor.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2019-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.azure.configuration;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+import com.azure.core.http.HttpMethod;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.management.profile.AzureProfile;
+
+import org.apache.commons.lang3.Validate;
+import org.vividus.azure.client.AzureResourceManagerClient;
+import org.vividus.azure.identity.CredentialFactory;
+import org.vividus.configuration.AbstractPropertiesProcessor;
+import org.vividus.util.json.JsonPathUtils;
+
+public class AzureKeyVaultPropertiesProcessor extends AbstractPropertiesProcessor
+{
+    private static final String ENVIRONMENT_PROPERTY = "azure.environment";
+    private static final String KEY_VAULT_NAME_PROPERTY = "azure.key-vault.name";
+    private static final String KEY_VAULT_API_VERSION_PROPERTY = "azure.key-vault.api-version";
+
+    private static final Map<String, AzureEnvironment> ENVIRONMENT_MAP = Map.of(
+            "AZURE", AzureEnvironment.AZURE,
+            "AZURE_CHINA", AzureEnvironment.AZURE_CHINA,
+            "AZURE_GERMANY", AzureEnvironment.AZURE_GERMANY,
+            "AZURE_US_GOVERNMENT", AzureEnvironment.AZURE_US_GOVERNMENT
+    );
+
+    private AzureEnvironment azureEnvironment;
+    private String keyVaultName;
+    private String apiVersion;
+    private AzureResourceManagerClient client;
+
+    public AzureKeyVaultPropertiesProcessor()
+    {
+        super("AZURE_KEY_VAULT");
+    }
+
+    @Override
+    public Properties processProperties(Properties properties)
+    {
+        this.azureEnvironment = ENVIRONMENT_MAP.get(properties.getProperty(ENVIRONMENT_PROPERTY));
```

Comment:
what if the property has invalid environment value?

Reply:
NPE, added AZURE as default in code